cmd/ctr/commands/run/run_unix.go

```diff
@@ -210,10 +249,51 @@ func NewContainer(ctx gocontext.Context, client *containerd.Client, context *cli
 }
 
 func getNewTaskOpts(context *cli.Context) []containerd.NewTaskOpts {
+	var (
+		tOpts []containerd.NewTaskOpts
+	)
 	if context.Bool("no-pivot") {
-		return []containerd.NewTaskOpts{containerd.WithNoPivotRoot}
+		tOpts = append(tOpts, containerd.WithNoPivotRoot)
+	}
+	if uidmap := context.String("uidmap"); uidmap != "" {
+		parts := strings.Split(uidmap, ":")
+		uid, _ := parseRemappingPart(parts[1], "uidmap", "uid")
+		tOpts = append(tOpts, containerd.WithUIDMap(uid))
+	}
+	if gidmap := context.String("gidmap"); gidmap != "" {
+		parts := strings.Split(gidmap, ":")
+		gid, _ := parseRemappingPart(parts[1], "gidmap", "gid")
+		tOpts = append(tOpts, containerd.WithGIDMap(gid))
+	}
+	return tOpts
+}
+
+func parseRemapping(remapping, idType string) (uint32, uint32, uint32, error) {
+	parts := strings.Split(remapping, ":")
+	if len(parts) != 3 {
+		return 0, 0, 0, errors.New("remapping user namespace using --" + idType + " requires the format 'container-id:id:size'")
+	}
+	container, err := parseRemappingPart(parts[0], idType, "container id")
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	id, err := parseRemappingPart(parts[1], idType, "id")
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	size, err := parseRemappingPart(parts[2], idType, "remapping size")
+	if err != nil {
+		return 0, 0, 0, err
+	}
+	return container, id, size, nil
+}
+
+func parseRemappingPart(part, idType, partName string) (uint32, error) {
```

Comment:
as noted in other review comments I think this function can be removed and simply grab the map struct from the `parseRemapping` function for use in the `TaskOpts` flow of trying to find the right uid/gid owners for IO.

Reply:
Sorry--I see it gets reused from `parseRemapping` as well; torn whether it makes sense as a separate function since I don't think it needs used from outside `parseRemapping` given my other comments, and if it's only used there the savings of having a call and `if err...` handler could pretty easily be replaced with the `strconv.ParseUint` and error handling directly in the `parseRemapping` function itself.

Reply:
Consolidated `parseRemappingPart()` into `parseIDMapping()`, addressed the error issues with `strconv.ParseUint`.